src/object/cli_obj.c

```diff
@@ -5336,6 +5330,13 @@ obj_update_shards_get(struct dc_object *obj, daos_obj_fetch_t *args, unsigned in
 		obj_shard = &obj->cob_shards->do_shards[shard_idx];
 		if (obj_shard->do_target_id == -1 || obj_shard->do_shard == -1 ||
 		    unlikely(DAOS_FAIL_CHECK(DAOS_FAIL_SHARD_NONEXIST))) {
+			/* check if the shard is from extending shard */
+			if (shard_idx % obj_get_grp_size(obj) > obj_ec_tgt_nr(obj_get_oca(obj))) {
```

Comment:
The shard_idx is started from zero, then should check ">= tgt_nr" ?

Reply:
ah, right. thanks.